client-web/src/exlibris-app/reducers/exlibris-reducer.html

```diff
@@ -39,9 +44,71 @@
                     //TODO:2018-55-17:yurii.haidamaka add data loading from the server
                     return state;
                 }
+                case ExlibrisApp.ExlibrisActions.FIND_BOOK_BY_ID: {
+                    // TODO:2018-06-11:yegor.udovchenko: Request book from the server by id
+                    return Object.assign({}, state, {
+                        bookDetails: {
+                            book: null,
+                            loading: true,
+                            notFound: false
+                        }
+                    });
+                }
+
+                case ExlibrisApp.ExlibrisActions.BOOK_RECEIVED: {
+                    // Searching for a book in a local data.
+                    // Should be removed when implement server response
+                    const id = action.bookId;
+                    const book = findElement(id, state.allBooks, state.expectedSoonBooks, state.borrowedBooks, state.reservedBooks);
```

Comment:
I think it will be better here.
```               
     const book = [...state.allBooks, ...state.expectedSoonBooks, ...state.borrowedBooks, ...state.reservedBooks].find((item) => item.id === id);

```